src/main/java/ca/corefacility/bioinformatics/irida/database/changesets/FastqcToFilesystem.java

```diff
@@ -0,0 +1,188 @@
+							+ chart + "' FROM analysis_output_file where execution_manager_file_id='" + chart
+							+ ".png'");
+		});
+
+		return new SqlStatement[0];
+	}
+
+	/**
+	 * Create the files in the output file directory and update the "analysis_output_file" entries
+	 *
+	 * @param chartType    the type of chart we're writing to the file system
+	 * @param jdbcTemplate {@link JdbcTemplate} to use for SQL
+	 */
+	private void writeFileForChartType(String chartType, JdbcTemplate jdbcTemplate) {
+
+		//going to do the updates in batches so things go smoothly
+		long batchsize = 10000;
+
+		//first get the count of analysis_fastqc entries we need to do for this chart type
+		String sql = "SELECT count(o.id) FROM analysis_fastqc f INNER JOIN analysis_output_file o ON f.id=o.analysis_id WHERE o.execution_manager_file_id=?";
+		Long entries = jdbcTemplate.queryForObject(sql, new Object[] { chartType + ".png" }, Long.class);
+		logger.info("Going to write " + entries + " entires for chart type " + chartType);
+
+		//doing this in a for loop to batch the entries
+		for (long offset = 0; offset < entries; offset += batchsize) {
+
+			//get a chunk of entries for this chart type
+			sql = "SELECT f.id, o.id, f." + chartType
+					+ " FROM analysis_fastqc f INNER JOIN analysis_output_file o ON f.id=o.analysis_id WHERE o.execution_manager_file_id=? limit "
+					+ batchsize + " offset " + offset;
+
+			List<Object[]> updates = jdbcTemplate.query(sql, new Object[] { chartType + ".png" },
+					//this mapper will look at the temp output file entry and write a file to the file system
+					new RowMapper<Object[]>() {
+						@Override
+						public Object[] mapRow(ResultSet rs, int rowNum) throws SQLException {
+
+							Long id = rs.getLong(1); //the analysis id
+							Long chartId = rs.getLong(2); //the analysis_output_file id
+							byte[] chart = rs.getBytes(3); //the chart type
+
+							//only writing progress for mod1000 results or the log gets crazy
+							if (id % 1000 == 0) {
+								logger.info("Mapping analysis " + id);
+							}
+
+							//get a path for <output files base dir>/<output file id>/1/
+							Path newFileDirectory = outputFileDirectory.resolve(chartId.toString())
+									.resolve("1");
+
+							try {
+								//create the directory
+								Files.createDirectories(newFileDirectory);
+								//get a path to the file
+								newFileDirectory = newFileDirectory.resolve(chartType + ".png");
+								//write the chart bytes to file
+								Files.write(newFileDirectory, chart);
+							} catch (IOException e) {
+								throw new SQLException("Couldn't create file", e);
+							}
+
+							// get the path as a string
+							String fullPath = newFileDirectory.toString();
+
+							//get a string representation of the output file directory
+							String basePath = outputFileDirectory.toString();
+
+							// relativize the path by stripping the base file path
+							fullPath = fullPath.replaceFirst(basePath + "/", "");
```

Comment:
Since everything is read and managed in `Path`s it shouldn't matter if the config contains a trailing slash.  Just to be sure I tested running the upgrade with a trailing slash and ended up with the same results (normal relativized paths).

Reply:
Thanks. I guess I did not notice that.

I checked the Java docs and it looks like there is a method [Path.relativize()](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#relativize-java.nio.file.Path-). Could you use that method instead of doing direct String matching?

Reply:
Yeah I had tried it in the past but had used it backwards so it was giving me a different result.  Using it properly now in 3735258 and it gives the correct result.  Better to work in `Path` space than `String` space as much as we can.